File: kotlin-bundled-compiler/src/com/intellij/formatting/KotlinSettingsProvider.java
package com.intellij.formatting;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.kotlin.idea.KotlinLanguage;
import org.jetbrains.kotlin.idea.core.formatter.KotlinCodeStyleSettings;

import com.intellij.lang.Language;
import com.intellij.openapi.options.Configurable;
import com.intellij.psi.codeStyle.CodeStyleSettings;
import com.intellij.psi.codeStyle.CodeStyleSettingsProvider;
import com.intellij.psi.codeStyle.CustomCodeStyleSettings;

public class KotlinSettingsProvider extends CodeStyleSettingsProvider {

    @Override
    public String getConfigurableDisplayName() {
        return KotlinLanguage.NAME;
    }

    @Override
    public Language getLanguage() {
        return KotlinLanguage.INSTANCE;
    }

    @Override
    public CustomCodeStyleSettings createCustomSettings(CodeStyleSettings settings) {
        return new KotlinCodeStyleSettings(settings);
    }

    @NotNull
    @Override
    public Configurable createSettingsPage(CodeStyleSettings settings, CodeStyleSettings originalSettings) {
        throw new UnsupportedOperationException();
    }
}

File: kotlin-bundled-compiler/src/com/intellij/formatting/KotlinSettingsProvider.kt
package com.intellij.formatting

import com.intellij.lang.Language
import com.intellij.openapi.options.Configurable
import com.intellij.psi.codeStyle.CodeStyleSettings
import com.intellij.psi.codeStyle.CodeStyleSettingsProvider
import com.intellij.psi.codeStyle.CustomCodeStyleSettings
import org.jetbrains.kotlin.idea.KotlinLanguage
import org.jetbrains.kotlin.idea.core.formatter.KotlinCodeStyleSettings

class KotlinSettingsProvider : CodeStyleSettingsProvider() {
    override fun getConfigurableDisplayName(): String = KotlinLanguage.NAME
    
    override fun getLanguage(): Language = KotlinLanguage.INSTANCE
    
    override fun createCustomSettings(settings: CodeStyleSettings): CustomCodeStyleSettings {
        return KotlinCodeStyleSettings(settings).apply {
            this.ALLOW_TRAILING_COMMA = true
            this.ALLOW_TRAILING_COMMA_ON_CALL_SITE = true
        }
    }
    
    override fun createSettingsPage(settings: CodeStyleSettings, originalSettings: CodeStyleSettings): Configurable {
        throw UnsupportedOperationException()
    }
}